src/main/java/mike/gui/Line.java

```diff
@@ -5,65 +5,117 @@
 import java.awt.Dimension;
 import java.awt.Graphics;
 import java.awt.Graphics2D;
+import java.awt.RenderingHints;
 import java.awt.geom.Line2D;
+import java.awt.geom.Path2D;
+
 import javax.swing.JComponent;
 import javax.swing.JLabel;
 
+import mike.datastructures.Relationship.Type;
+
 public class Line extends JComponent {
 	private static final long serialVersionUID = 1L;
 	double x1, y1, x2, y2;
 	JLabel classOne, classTwo;
-	Line(double x1, double y1, double x2, double y2, JLabel classOne, JLabel classTwo) {
+	boolean toSelf;
+	Type type;
+
+	Line(double x1, double y1, double x2, double y2, JLabel classOne, JLabel classTwo, Type type) {
 		super();
-		
+		this.type = type;
+		this.toSelf = toSelf;
 		this.x1 = x1;
 		this.y1 = y1;
 		this.x2 = x2;
 		this.y2 = y2;
 		this.classOne = classOne;
 		this.classTwo = classTwo;
-		
+
 		this.setVisible(true);
-		this.setPreferredSize(new Dimension (782, 725));
+		this.setPreferredSize(new Dimension(classOne.getParent().getWidth(), classOne.getParent().getHeight()));
 	}
-	
+
 	public void setNewPoints(double newx1, double newy1, double newx2, double newy2) {

```

Comment:
If you pass in the classes so that the line has a reference, can't this be replaced with an update() method that determines the new point locations internally?